src/app/bundle.ts

```diff
@@ -12,41 +12,32 @@ export function createPatientBundle(
   parameters: { [key: string]: Stringable },
   entries: fhirclient.FHIR.BundleEntry[]
 ): string {
-  let paramResource = `{
-                         "resourceType": "Parameters",
-                         "id": "0",
-                         "parameter": [`;
+  const paramResource: ParameterResource = { resourceType: 'Parameters', id: '0', parameter: [] };
   for (const p in parameters) {
     // Ignore null values, they indicate a parameter that should be ignored.
     const value = parameters[p];
     if (value === null) continue;
-    paramResource += `{
-                        "name": ${JSON.stringify(p)},
-                        "valueString": ${JSON.stringify(value)}
-                      },`;
+    paramResource.parameter.push({ name: p, valueString: value });
   }
-  // need to remove final comma
-  paramResource = paramResource.slice(0, -1);
-  paramResource += `
-                     ]
-                    }
-                  },`;
-  let patientBundle =
-    `{
-                          "resourceType": "Bundle",
-                          "type": "collection",
-                          "entry": [
-                            {
-                              "resource": ` + paramResource;
-  //for (const resource in entries)
+  const patientBundle: PatientBundle = {
+    resourceType: 'Bundle',
+    type: 'collection',
+    entry: [{ resource: paramResource }]
+  };
   entries.forEach((resource) => {
-    // for each instead
-    patientBundle += `
-    ${JSON.stringify({ fullUrl: resource.fullUrl, resource: resource.resource })},`;
+    patientBundle.entry.push({ fullUrl: resource.fullUrl, resource: resource.resource });
   });
-  patientBundle = patientBundle.slice(0, -1);
-  patientBundle += `
-                      ]
-                     }`;
-  return patientBundle;
+  return JSON.stringify(patientBundle);
```

Comment:
We don't want to stringify the JSON object at the end. We want to be returning the JSON object.